pkg/bitcoin/chain.go

```diff
@@ -1,5 +1,12 @@
 package bitcoin
 
+// Config contains the config data needed to connect to the Bitcoin chain.
+type Config struct {
+	URL      string
+	Username string
+	Password string
+}
```

Comment:
This is the wrong place for that. This config is implementation-specific and should be exposed by `bitcoin.Client` implementations.

Reply:
I wanted to put this `Config` struct, so that we have a clean `bitcoin.Config` field in the general client configuration that any possible command of `keep-client` could reuse:
```
// Config is the top level config structure.
 type Config struct {
 	Ethereum   commonEthereum.Config
 	Bitcoin    bitcoin.Config
 	LibP2P     libp2p.Config `mapstructure:"network"`
 	Storage    storage.Config
 	ClientInfo clientinfo.Config
 	Maintainer maintainer.Config
 	Tbtc       tbtc.Config
 }
 ```
 
 My reasoning was that `URL`, `username` and`password` fields inside bitcoin.Config` are pretty generic and will fit any bitcoin implementation.
 
Should I move this config into the electrum package, so that we have 
```Electrum   electrum.Config```
inside the client config?

Reply:
The `pkg/bitcoin` should only contain types from the Bitcoin protocol specification (see package documentation). There is no such type as `Config` in the Bitcoin spec. This is a pure implementation-specific concept and it should live in the implementation package. Also, different implementations could have different configs, one of them can use a username and password but another one can use an authentication token. This is another argument that we should keep the config on the implementation level. Last but not least, note that the `Ethereum commonEthereum.Config` is also implementation-specific, we don't declare it on the `pkg/chain` level. So as `LibP2P libp2p.Config` which is libp2p-specific and is not declared on `pkg/net` level. That being said, I think we should declare `Electrum electrum.Config` just as you proposed.

Reply:
Replaced `Bitcoin` configuration with `Electrum` in 2eb93eaefef4c21cbbcb13e47a1f1c465fe81576.
Once https://github.com/keep-network/keep-core/pull/3395 is merged to main, we will replace `Electrum` with `Electrs`. For now, let's stick to `Electrum`, as that's the name we have in the base branch of my PR.

Reply:
> For now, let's stick to Electrum, as that's the name we have in the base branch of my PR.

We can change it even now. The `pkg/bitcoin/electrum` was just supposed to demonstrate the intended package structure. Maybe we can coordinate with @nkuba and change the config accordingly in this PR to have that covered?

Reply:
I added the same Config as @nkuba in #3395 in 9152aed1f0c75575d09f60d66bb71ffe35a558d7.
Updated all the code changed by this PR accordingly.
@nkuba should now have conflicts with this PR.